src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/commands.py

```diff
@@ -237,36 +242,37 @@ def transform_av_set_collection_output(av_sets):
 # VM Size
 cli_command(__name__, 'vm list-sizes', mgmt_path.format('virtual_machine_sizes_operations', 'VirtualMachineSizesOperations', 'list'), cf_vm_sizes)
 
-# VM Disk
-op_var = 'disks_operations'
-op_class = 'DisksOperations'
-cli_command(__name__, 'disk create', custom_path.format('create_managed_disk'))
-cli_command(__name__, 'disk list', custom_path.format('list_managed_disks'))
-cli_command(__name__, 'disk show', mgmt_path.format(op_var, op_class, 'get'), cf_disks, exception_handler=empty_on_404)
-cli_command(__name__, 'disk delete', mgmt_path.format(op_var, op_class, 'delete'), cf_disks)
-cli_command(__name__, 'disk grant-access', custom_path.format('grant_disk_access'))
-cli_command(__name__, 'disk revoke-access', mgmt_path.format(op_var, op_class, 'revoke_access'), cf_disks)
-cli_generic_update_command(__name__, 'disk update', 'azure.mgmt.compute.operations.{}#{}.get'.format(op_var, op_class),
-                           'azure.mgmt.compute.operations.{}#{}.create_or_update'.format(op_var, op_class),
-                           custom_function_op=custom_path.format('update_managed_disk'),
-                           setter_arg_name='disk', factory=cf_disks)
-
-op_var = 'snapshots_operations'
-op_class = 'SnapshotsOperations'
-cli_command(__name__, 'snapshot create', custom_path.format('create_snapshot'))
-cli_command(__name__, 'snapshot list', custom_path.format('list_snapshots'))
-cli_command(__name__, 'snapshot show', mgmt_path.format(op_var, op_class, 'get'), cf_snapshots, exception_handler=empty_on_404)
-cli_command(__name__, 'snapshot delete', mgmt_path.format(op_var, op_class, 'delete'), cf_snapshots)
-cli_command(__name__, 'snapshot grant-access', custom_path.format('grant_snapshot_access'))
-cli_command(__name__, 'snapshot revoke-access', mgmt_path.format(op_var, op_class, 'revoke_access'), cf_snapshots)
-cli_generic_update_command(__name__, 'snapshot update', 'azure.mgmt.compute.operations.{}#{}.get'.format(op_var, op_class),
-                           'azure.mgmt.compute.operations.{}#{}.create_or_update'.format(op_var, op_class),
-                           custom_function_op=custom_path.format('update_snapshot'),
-                           setter_arg_name='snapshot', factory=cf_snapshots)
-
-op_var = 'images_operations'
-op_class = 'ImagesOperations'
-cli_command(__name__, 'image create', custom_path.format('create_image'))
-cli_command(__name__, 'image list', custom_path.format('list_images'))
-cli_command(__name__, 'image show', mgmt_path.format(op_var, op_class, 'get'), cf_images, exception_handler=empty_on_404)
-cli_command(__name__, 'image delete', mgmt_path.format(op_var, op_class, 'delete'), cf_images)
+
+if get_api_version(ResourceType.MGMT_COMPUTE) in ['2016-04-30-preview']:
```

Comment:
VM and VMSS create don't fully support older API versions.
This would be part of https://github.com/Azure/azure-cli/issues/2272.